src/nsidc/metgen/collection_metadata.py

```diff
@@ -0,0 +1,268 @@
+
+        # Validate and parse response
+        ummc = self._validate_cmr_response(cmr_response, short_name, version_str)
+
+        return self._parse_ummc_to_metadata(ummc, short_name, version_str)
+
+    def _validate_cmr_response(
+        self, response: list, short_name: str, version: str
+    ) -> dict:
+        """
+        Validate the CMR response and extract the UMM-C record.
+
+        Args:
+            response: Raw response from earthaccess
+            short_name: Collection short name for error messages
+            version: Collection version for error messages
+
+        Returns:
+            Validated UMM-C dictionary
+
+        Raises:
+            ValueError: If response is invalid
+        """
+        if not response:
+            raise ValueError(
+                f"Empty UMM-C response from CMR for {short_name}.{version}"
+            )
+
+        if len(response) > 1:
+            raise ValueError(
+                f"Multiple UMM-C records returned from CMR for {short_name}.{version}, "
+                "none will be used."
+            )
+
+        # Check that the response item is a dict before extracting
+        if not isinstance(response[0], dict) or "umm" not in response[0]:
+            raise ValueError(
+                f"No UMM-C content in CMR response for {short_name}.{version}"
+            )
+
+        # Extract the UMM-C content
+        ummc = response[0].get("umm", response[0])
+
+        if not isinstance(ummc, dict):
+            raise ValueError(
+                f"Invalid UMM-C format in CMR response for {short_name}.{version}"
+            )
+
+        return ummc
+
+    def _parse_ummc_to_metadata(
+        self, ummc: dict, short_name: str, version: str
+    ) -> CollectionMetadata:
+        """
+        Parse UMM-C record into structured metadata.
+
+        Args:
+            ummc: UMM-C dictionary from CMR
+            short_name: Collection short name
+            version: Collection version
+
+        Returns:
+            Populated CollectionMetadata object
+        """
+        # Extract temporal extent and check for errors
+        temporal_extent, temporal_error = self._parse_temporal_extent(ummc)
+
+        # Build the metadata object
+        return CollectionMetadata(
+            short_name=short_name,
+            version=version,
+            entry_title=ummc.get("EntryTitle", f"{short_name}.{version}"),
+            granule_spatial_representation=self._extract_nested_value(
+                ummc, constants.GRANULE_SPATIAL_REP_PATH
+            ),
+            spatial_extent=self._extract_nested_value(
+                ummc, constants.SPATIAL_EXTENT_PATH
+            ),
+            temporal_extent=temporal_extent,
+            temporal_extent_error=temporal_error,
+            processing_level_id=ummc.get("ProcessingLevel", {}).get("Id"),
+            collection_data_type=ummc.get("CollectionDataType"),
+            raw_ummc=ummc,
+        )
+
+    def _parse_temporal_extent(
```

Comment:
I kinda like test coverage that hits the individual pieces, maybe because it's a whole lot easier to set up a test for the thing that handles the temporal metadata than it is to test something like `_parse_ummc_to_metadata`.  Is this approach I see in `test_get_collection_metadata_success` recommended by your robot assistant or are you simply masochistic?